fedimint-server/src/net/api.rs

```diff
@@ -181,8 +139,60 @@ fn attach_endpoints<M>(
     }
 }
 
-fn server_endpoints() -> &'static [ApiEndpoint<FedimintConsensus>] {
-    const ENDPOINTS: &[ApiEndpoint<FedimintConsensus>] = &[
+fn attach_endpoints_erased(
+    rpc_module: &mut RpcModule<RpcHandlerCtx>,
+    server_module: &ServerModule,
+) {
+    let base_name = server_module.api_base_name();
+    let endpoints = server_module.api_endpoints();
+    let module_key = server_module.module_key();
+
+    for endpoint in endpoints {
+        // This memory leak is fine because it only happens on server startup
+        // and path has to live till the end of program anyways.
+        let path: &'static _ =
+            Box::leak(format!("/{}{}", base_name, endpoint.path).into_boxed_str());
+        let handler: &'static _ = Box::leak(endpoint.handler);
+
+        rpc_module
+            .register_async_method(path, move |params, state| {
+                Box::pin(async move {
+                    let params = params.one::<serde_json::Value>()?;
+                    // Using AssertUnwindSafe here is far from ideal. In theory this means we could
```

Comment:
BTW. the whole `UnwindSafe` is one of a few broken things in Rust and possibly will be deprecated altogether. https://github.com/rust-lang/rust/issues/40628

Reply:
I think it's a separate issue (created #943), I just copied the old code to adapt it.